scipy/stats/_continuous_distns.py

```diff
@@ -5508,67 +5508,119 @@ def _entropy(self, s):
-            raise ValueError("All parameters fixed. There is nothing to "
-                             "optimize.")
+        def get_shape_scale(loc):
+            if fshape is None or fscale is None:
+                lndata = np.log(data - loc)
+            scale = fscale or np.exp(lndata.mean())
+            shape = fshape or np.sqrt(np.mean((lndata - np.log(scale))**2))
+            return shape, scale
 
-        data = np.asarray(data)
+        def dL_dLoc(loc):
+            shape, scale = get_shape_scale(loc)
+            shifted = data - loc
+            return np.sum((1 + np.log(shifted/scale)/shape**2)/shifted)
 
-        if not np.isfinite(data).all():
-            raise ValueError("The data contains non-finite values.")
+        def get_estimate(loc):
+            shape, scale = get_shape_scale(loc)
+            return shape, loc, scale
 
-        floc = float(floc)
-        if floc != 0:
-            # Shifting the data by floc. Don't do the subtraction in-place,
-            # because `data` might be a view of the input array.
-            data = data - floc
-        if np.any(data <= 0):
-            raise FitDataError("lognorm", lower=floc, upper=np.inf)
-        lndata = np.log(data)
-
-        # Three cases to handle:
-        # * shape and scale both free
-        # * shape fixed, scale free
-        # * shape free, scale fixed
+        parameters = _check_fit_input_parameters(self, data, args, kwds)
+        data, fshape, floc, fscale = parameters
+        mn = data.min()
 
-        if fscale is None:
-            # scale is free.
-            scale = np.exp(lndata.mean())
-            if f0 is None:
-                # shape is free.
-                shape = lndata.std()
+        if floc is None:
+            mn_inf = np.nextafter(mn, -np.inf)
+            with np.errstate(over='raise'):
+                try:
+                    std = data.std()
+                except FloatingPointError:
+                    # Handle possible overflows when squaring.
+                    absmx = np.max(np.abs(data))
+                    std = np.std(data/absmx) * absmx
+            # Range of acceptable locations based on sample statistics.
+            lbrack, rbrack = data.mean() - alpha_std*std, mn_inf
+
+            i = 1.
+            delta = mn - mn_inf
+            while dL_dLoc(rbrack) >= 0:
+                i *= 2
+                rbrack = mn - delta*i
+
+            if not (rbrack > lbrack):
+                loc = mn_inf
             else:
-                # shape is fixed.
-                shape = float(f0)
+                # If the derivative is positive at `mn_inf`, and negative a few
+                # `delta` away from it, even though root finding will work, due
+                # to numerical inaccuracy the derivative evaluated at the root
+                # will be much too far away from zero, so that solution is
+                # discarded.
+                has_right_divergence = i > 1
```

Comment:
This determines whether to consider `np.inf(data.min(), -np.inf)` as the estimated location, if roots are found on the interval.